src/PerformanceTests/Utils/Powerplan.cs

```diff
@@ -1,9 +1,40 @@
 using System;
 using System.Runtime.InteropServices;
+using System.Threading;
+using log4net;
 
 public class Powerplan
 {
-    public static Guid GetActive()
+    static ILog Log = LogManager.GetLogger(typeof(Powerplan));
+
+    public static void CheckPowerPlan()
```

Comment:
I'm fine with moving this, although this method doesn't really belong to the Powerplan class.

Reply:
I partially agree :D In general, the check should indeed not belong to the powerplan class, but as long as the check contains code like this:
> var highperformance = new Guid("8c5e7fda-e8bf-4a96-9a85-a6e23a8c635c");
it's leaking too many internals and I don't think this should be outside the powerplan class. If the Powerplan class would have a property like `Powerplan.IsSetToHighPerformance` which can easily be checked without requiring knowledge of magic guids, I'd totally agree with you. For now, I'd suggest to keep as is and see whether we can get a round of refactorings in before closing the issue.